src/main/java/uk/co/ramp/event/EventList.java

```diff
@@ -0,0 +1,83 @@
+package uk.co.ramp.event;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import uk.co.ramp.event.types.ContactEvent;
+import uk.co.ramp.event.types.Event;
+import uk.co.ramp.event.types.ImmutableFormattedEvent;
+import uk.co.ramp.io.csv.CsvException;
+import uk.co.ramp.io.csv.CsvWriter;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Repository
+public class EventList {
+
+    private final Map<Integer, List<Event>> map = new HashMap<>();
+    private final Map<Integer, List<Event>> completedMap = new HashMap<>();
+    private final FormattedEventFactory formattedEventFactory;
+
+    private static final Logger LOGGER = LogManager.getLogger(EventList.class);
+
+    @Autowired
+    public EventList(FormattedEventFactory formattedEventFactory) {
+        this.formattedEventFactory = formattedEventFactory;
+    }
+
+    //create
+    public void addEvent(Event e) {
+        map.computeIfAbsent(e.time(), k -> new ArrayList<>()).add(e);
+    }
+
+    public void addEvents(List<Event> events) {
+        events.forEach(this::addEvent);
+    }
+
+    public void addEvents(Map<Integer, List<ContactEvent>> readEvents) {
+
+        for (Map.Entry<Integer, List<ContactEvent>> entry : readEvents.entrySet()) {
+            bulkAdd(entry.getKey(), entry.getValue());
+        }
+    }
+
+    private void bulkAdd(int time, List<ContactEvent> events) {
+        map.computeIfAbsent(time, k -> new ArrayList<>()).addAll(events);
+    }
+
+    public void completed(Event e) {
+        completedMap.computeIfAbsent(e.time(), k -> new ArrayList<>()).add(e);
+    }
+
+
+    //read
+    public List<Event> getForTime(int time) {
+        return map.getOrDefault(time, new ArrayList<>());
+    }
+
+    public Map<Integer, List<Event>> getMap() {
+        return map;
+    }
```

Comment:
It is unnecessary to expose the map publicly. The only caller of this is from outbreak where outbreak.java is computing the last contact. We can replace this method with the following:
```suggestion
    public Optional<Integer> lastContactTime() {
        return map.keySet().stream().max(Comparator.naturalOrder());
    }
```

and in Outbreak.java we can call:
`int lastContact = eventList.lastContactTime().orElseThrow();`
instead of 
`int lastContact = eventList.getMap().keySet().stream().max(Comparator.naturalOrder()).orElseThrow();`

Reply:
good call